.github/workflows/py-build-main.yml

```diff
@@ -20,11 +20,11 @@ jobs:
         python-version: [3.7, 3.8, 3.9, "3.10", 3.11]
         os: [ubuntu-latest, windows-latest, macos-latest]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           submodules: true
       - name: setup python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
```

Comment:
SWE recommended me using `actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c` which is the latest version.

Reply:
Should we be doing that across the board?  Do we trust these major mainstream actions  like `actions/checkout` or `actions/setup-python` enough?  I felt this was more relevant for lesser-known actions.

Reply:
Maybe @griffinfujioka can help with this one.